GatherPress/gatherpress
# new calendar endpoints | part 4/4 | endpoint implementations

This PR is **the final part** of the fragmentation of #831, which was way too big to review or merge.
*I try to do better in the future.*

Finally closes #603



### Description of the Change


**This PR introduces the following of new endpoints:**

- `example.org/event/my-sample-event/ical`

   provides a download-able .ics file in ical format.

- `example.org/event/my-sample-event/outlook`

   provides the same download-able file as an alias.

- `example.org/event/my-sample-event/google-calendar`

   redirects to create a new event in *Google Calendar*.

- `example.org/event/my-sample-event/yahoo-calendar`

   redirects to create a new event in *Yahoo Calendar*.

- `example.org/event/feed/ical`

   provides a subscribe-able event feed in ical format with all events of the site.

- `example.org/venue/my-sample-venue/feed/ical`

   provides a subscribe-able event feed in ical format with all events at that venue.

- `example.org/topic/my-sample-topic/feed/ical`

   provides a subscribe-able event feed in ical format with all events grouped into that topic.


### How to test the Change


- Test the _Add to Google calendar_
   1. (You need an google account & use their calendar)
   2. Go to the frontend of any event and hit the the **Add to calendar > Google** button
   3. Confirm that the drafted event in the Google calendar matches the GatherPress event
- Test the _Add to Yahoo calendar_
   1. (You need an yahoo account & use their calendar)
   2. Go to the frontend of any event and hit the the **Add to calendar > Yahoo** button
   3. Confirm that the drafted event in the Yahoo calendar matches the GatherPress event
- …
- create a ical-download.php file and put it at the root of your active theme to overwrite the plugin template 
   1. Create that file
   5. Put some `echo ‚hello themed iCal world‘;` into it
   6. Go to the frontend of any event and hit the the **Add to calendar > iCal** button (to make use of your template)
- https://icalendar.org/validator.html

### Changelog Entry

> Added - New URL endpoints for subscribe-able ical feeds for events, venues & topics

### Credits

Props @carstingaxion

### Checklist:




#### Related PRs that can be reviewed & merged after this:

- https://github.com/GatherPress/gatherpress-awesome/pull/3
- https://github.com/carstingaxion/gatherpress/pull/73

#### To get the new endpoints into the block-editor, one of the following would be required:

- https://github.com/WordPress/gutenberg/issues/65351
   _OR_ 
- https://github.com/carstingaxion/gatherpress-feed-buttons
